### DISCUSSION TITLE
Kubernetes deployment

### DISCUSSION THREAD

**Discussion by mth3r** [investbrainapp/investbrain#27] [Q&A] [answered]
I'd like to deploy this in my cluster, but I cant seem to get the docker image to work.  Any suggestions on how to set the container image?

**Comment by hackeresq** [comment]
Unfortunately, not very familiar with Kubernetes.

Perhaps the issue is due to the way the database is provisioned?  You may need to update (or customize) the compose file to be compatible.

Were there any error messages you can share? Would be helpful to troubleshoot.

**Comment by hackeresq** [comment]
Going to close this discussion for now since Kubernetes support seems like a unique request. Happy to re-open if there's additional interest!

**Comment by hackeresq** [comment]
This has been discussed multiple times (see #9).

This will need to be something added by the community, as I don't have a K8s environment to test this with. This should be possible with the recent release (uses separate Docker image).

**Reply by Marx2** [reply]
You don't need any special k8s environment. You can easily run k8s locally, for example through docker UI, or minikube.

**Comment by Marx2** [comment]
docker:latest still produces: 
exec /bin/bash: exec format error
Looks like entrypoint is invalid

**Comment by hackeresq** [comment, marked as answer]
@mth3r, @Marx2 - I've just tagged a new release which I suspect will be more compatible with K8s. Want to give that a shot? 

Edit: https://github.com/investbrainapp/investbrain/releases/tag/v1.1.0

**Reply by Marx2** [reply]
I've tested the newest 1.1.4. It works much better, but I have some problem. Investbrain seems to start ok in Kubernetes, but then it dies and restart
Logs doesn't say much. Can I enable more debug level in some way?

```
2025-02-02 22:40:07,779 INFO spawned: 'scheduler' with pid 21

  ********************************************************************************
  *  III   N   N  V   V  EEEEE  SSSS  TTTTT  BBBBB   RRRR    AAAAA  III   N   N  *
  *   I    NN  N  V   V  E      S       T    B    B  R   R   A   A   I    NN  N  *
  *   I    N N N  V   V  EEEE   SSSS    T    BBBBB   RRRR    AAAAA   I    N N N  *
  *   I    N  NN  V   V  E          S   T    B    B  R  R    A   A   I    N  NN  *
  *  III   N   N   VVV   EEEEE  SSSS    T    BBBBB   R   R   A   A  III   N   N  *
  ********************************************************************************
  
====================== Validating environment...  ====================== 

 > Storage directory scaffolding is OK... 

 > APP_KEY is OK... 

====================== Running migrations...  ======================

   INFO  Nothing to migrate.  

====================== Spinning up Supervisor daemon...  ====================== 

2025-02-02 22:43:05,521 INFO Set uid to user 0 succeeded
2025-02-02 22:43:05,530 INFO supervisord started with pid 1
2025-02-02 22:43:06,535 INFO spawned: 'nginx' with pid 17
2025-02-02 22:43:06,540 INFO spawned: 'php' with pid 18
2025-02-02 22:43:06,546 INFO spawned: 'queue-worker_00' with pid 19
2025-02-02 22:43:06,551 INFO spawned: 'queue-worker_01' with pid 20
2025-02-02 22:43:06,556 INFO spawned: 'scheduler' with pid 21
```

Here are my environment variables:
https://github.com/Marx2/homelab/blob/30781451f01bcf8717ee176cdfd84ce290ed5415/cluster/apps/pfire/investbrain/app/helmrelease.yaml#L42-L57

**Comment by mth3r** [comment]
I was able to deploy it without errors.  When I go through port forwarding to view the app, the layout is wonky.  Like the icons are huge and the text is unreadable.  I'll try to go through a service and see if that changes anything, but I imagine it has something to do with nginx, but that is just a guess.
________________________________

**Comment by hackeresq** [comment]
Are you getting any errors in the browser console? Particularly "unsafe content blocked" or "not found" errors? This has happened to other folks - they had to update the `ASSET_URL` environment variable.

**Comment by mth3r** [comment]
i did as you suggested and it works. thanks.  I'm getting some other permission errors, but I think that is a vestiage of doing multiple deployments and not clearing out the volumes.  Looks great and thanks for your efforts

________________________________

**Reply by hackeresq** [reply]
So glad to hear it! 

Let me know if you continue having volume issues.

**Reply by hackeresq** [reply]
There were others with the permission errors on the releases from earlier this week. I'm fairly confident those are solved starting with v1.1.4. Would be interested if you could confirm this is working now?

**Reply by Marx2** [reply]
I should add that the problem for me was not enough resources. After giving more RAM container starts ok in Kubernetes
